arcade/arcade/cli/main.py

```diff
@@ -177,38 +174,7 @@ def show(
     """
     Show the available toolkits or detailed information about a specific tool.
     """
-    try:
-        if local:
-            catalog = create_cli_catalog(toolkit=toolkit)
-            tools = [t.definition for t in list(catalog)]
-        else:
-            tools = get_tools_from_engine(host, port, force_tls, force_no_tls, toolkit)
-
-        if tool:
-            # Display detailed information for the specified tool
-            tool_def = next(
-                (
-                    t
-                    for t in tools
-                    if t.get_fully_qualified_name().name.lower() == tool.lower()
-                    or str(t.get_fully_qualified_name()).lower() == tool.lower()
-                ),
-                None,
-            )
-            if not tool_def:
-                console.print(f"❌ Tool '{tool}' not found.", style="bold red")
-                typer.Exit(code=1)
-            else:
-                display_tool_details(tool_def)
-        else:
-            # Display the list of tools as a table
-            display_tools_table(tools)
-
-    except Exception as e:
-        if debug:
-            raise
-        error_message = f"❌ Failed to list tools: {escape(str(e))}"
-        console.print(error_message, style="bold red")
+    show_logic(toolkit, tool, host, local, port, force_tls, force_no_tls, debug)
```

Comment:
Moved logic to a wrapper function so that the CLI command can be called directly without any unintended consequences such as unresolved parameters.